src/lexer/tokens.rs

```diff
@@ -298,6 +298,9 @@ pub enum Token {
     #[token("POINTER", ignore(case))]
     KeywordPointer,
 
+    #[token("FNPTR", ignore(case))]
```

Comment:
Happy to change this if there are "better" suggestions. Also due to integration tests the token now leaks to the end-user although function pointers are a internal-only construct as of now. No idea what we want to do here tbh. The easiest solution would be a validation error but disabling it in tests (if possible)?

Reply:
we could always just not parse it and have it as an ast only construct?
Also just a nit, but FPOINTER? FUNCT_REF? FUNCT_PTR?

Reply:
How would you test if code execution is correct then? Specifically, we need to parse them if you take a look at the integration / lit tests, no?

Reply:
since not intended for the user, we test through polymorphism.. but I don't mind them here, just be careful about reserving new keywords: I would go with `__FPOINTER` otherwise any variable called FPOINTER or FNPTR in the old version would no longer work

Reply:
Yeah, I would like to explicitly test them since that should (in theory) make debugging bugs related to function pointers easier rather than opaquely couple them to polymorphism. Changed `FNPTR` to `__FPOINTER`